kuma/javascript/src/live-examples.js

```diff
@@ -48,20 +48,14 @@ export function addLiveExampleButtons(rootElement) {
         // direct siblings and also descendants of direct siblings
         // (because sometimes some of the source code is tucked inside
         // a hidden div).
-        let html = rootElement.querySelector(
-            `#${sectid} ~ pre[class*=html], #${sectid} ~ * pre[class*=html]`
-        );
-        let css = rootElement.querySelector(
-            `#${sectid} ~ pre[class*=css], #${sectid} ~ * pre[class*=css]`
-        );
-        let js = rootElement.querySelector(
-            `#${sectid} ~ pre[class*=js], #${sectid} ~ * pre[class*=js]`
-        );
+        let html = section.querySelectorAll('pre[class*=html]');
+        let css = section.querySelectorAll('pre[class*=css]');
+        let js = section.querySelectorAll('pre[class*=js]');
 
         // Now get the source code out of those pre elements
-        let htmlCode = html ? html.textContent : '';
-        let jsCode = js ? js.textContent : '';
-        let cssCode = css ? css.textContent : '';
+        let htmlCode = [...html].map(p => p.textContent).join('');
+        let jsCode = [...js].map(p => p.textContent).join('');
+        let cssCode = [...css].map(p => p.textContent).join('');
```

Comment:
This is so clean and nice to read, my only concern is IE :( Perhaps doing it more IE friendly would be the better option, thanks @peterbe

Reply:
I don't get it. Why wouldn't this work in IE?

Reply:
The spread operator is not supported in IE - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Browser_compatibility

and neither is arrow functions - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Browser_compatibility

Perhaps I am not thinking about it correctly?

Reply:
Hehe. You forgot that this is converted to ES5  by Babel :)

Reply:
Doh! For some reason, I had it in my head that this was part of the old JS :p